docs/adr/0006-process-for-onboarding-new-tenants.md

```diff
@@ -0,0 +1,299 @@
+# Process for onboarding new tenants and enrolling them to use Marain services
+
+## Status
+
+Proposed
+
+## Context
+
+We have defined (in ADR 0006) the way in which we intend to implement Tenancy in Marain instances using the `Marain.Tenancy` service. As noted in that ADR, managing the desired model by hand would be excessively error prone and as such, we need to design tooling that will allow us to create and manage new tenants, and to allow them to use the Marain services they are licenced for.
+
+Before we can build that tooling we need to design the underlying process by which tenant onboarding, enrollment and offboarding will work. This needs to allow new Client Tenants to be onboarded into Marain services without tightly coupling the services so some central thing that knows everything about them.
+
+## Decision
+
+We are envisaging a central control-plane API (referred to for the remainder of this document as the "Management API") for Marain which primarily builds on top of the Tenancy service. This will provide the standard operations such as creating new tenants and enrolling them to use Marain services.
+
+It will also need to allow us to manage licencing, but we anticipate that this will be via an as-yet nonexistent API, as it will be a central service much like tenancy, used by all other services (either directly or via something like APIM).
+
+### Onboarding
+
+Onboarding is a relatively simple part of the process where we create a new Tenant for the client. We will need to determine how we intend licencing to work and what part, if any, the Management API plays..
+
+### Enrollment
+
+Service enrollment is a more interesting aspect of the process. In order to avoid tightly coupling Marain services to the Management API, we need two things:
+- A means of discovering the available services.
+- A means of requesting that a service enroll a Tenant to use it.
+
+As described in ADR 0005, we are envisaging that each service has a Tenant created for it, under a single parent for all Service Tenants. These tenants can then underpin the discovery mechanism that allows the management API to enumerate services that tenants can be enrolled into. Part of the data attached to each tenant will be the URIs at which the endpoints they expose can be found.
+
+Once we have provided a discovery mechanism, we need to define a way in which a service can be asked to enroll a tenant. The simplest way for this to work is to require all Marain services to have a control plane API that implements a set of standard endpoints. We can provide a standard OpenAPI document that defines the endpoints needed, which we would currently expect to be:
```

Comment:
Isn't this the sort of scenario you want to use pub / sub / event grid for? Broadcast a "provision" event and all services subscribe and enroll the tenant?